nautobot/ipam/tests/test_forms.py

```diff
@@ -1,39 +1,82 @@
 """Test IPAM forms."""
+
 from django.test import TestCase
 
 from nautobot.extras.models.statuses import Status
-from nautobot.ipam.forms import IPAddressForm
+from nautobot.ipam import forms, models
+
 
+class BaseNetworkFormTest:
+    form_class = None
+    field_name = None
+    object_name = None
+    extra_data = {}
 
-class IPAddressFormTest(TestCase):
     def test_valid_ip_address(self):
-        form = IPAddressForm(data={"address": "192.168.1.0/24", "status": Status.objects.get(slug="dhcp")})
-        self.assertTrue(form.is_valid())
-        self.assertTrue(form.save())
+        data = {self.field_name: "192.168.1.0/24", "status": Status.objects.get(slug="active")}
+        data.update(self.extra_data)
+        form = self.form_class(data)
 
-    def test_slaac_valid_ipv6(self):
-        form = IPAddressForm(
-            data={"address": "2001:0db8:0000:0000:0000:ff00:0042:8329/128", "status": Status.objects.get(slug="slaac")}
-        )
         self.assertTrue(form.is_valid())
         self.assertTrue(form.save())
 
-    def test_slaac_status_invalid_ipv4(self):
-        form = IPAddressForm(data={"address": "192.168.0.1/32", "status": Status.objects.get(slug="slaac")})
-        self.assertFalse(form.is_valid())
-        self.assertEquals("Only IPv6 addresses can be assigned SLAAC status", form.errors["status"][0])
-
     def test_address_invalid_ipv4(self):
-        form = IPAddressForm(data={"address": "192.168.0.1/64", "status": Status.objects.get(slug="dhcp")})
+        data = {self.field_name: "192.168.0.1/64", "status": Status.objects.get(slug="active")}
+        data.update(self.extra_data)
+        form = self.form_class(data)
+
         self.assertFalse(form.is_valid())
-        self.assertEquals("Please specify a valid IPv4 or IPv6 address.", form.errors["address"][0])
+        self.assertEquals("Please specify a valid IPv4 or IPv6 address.", form.errors[self.field_name][0])
```

Comment:
Nit: `assertEquals` is a deprecated alias of `assertEqual` - mind changing these to `assertEqual` throughout?

Reply:
TIL. Done.